rpcserver.go

```diff
@@ -2728,11 +2727,68 @@ func (r *rpcServer) GetNetworkInfo(ctx context.Context,
 		minChannelSize = 0
 	}
 
-	// TODO(roasbeef): graph diameter
+	// Calculate graph diameter: Diameter = maximum length of shortest path
+	// in graph between two nodes.
+	var diam int
+
+	// We'll run through all the nodes in graph, find farthest node from
+	// each, and take the maximum of that.
+	if err := graph.ForEachNode(nil, func(tx *bolt.Tx, SourceNode *channeldb.LightningNode) error {
+
+		// This will store the farthest node
+		var maxdist = -1
+		current := []*channeldb.LightningNode{SourceNode}
+		visited := map[string]bool{}
+
+		visited[hex.EncodeToString(SourceNode.PubKey.SerializeCompressed())] = true
+
+		// current at all times will store the current layer of nodes we
+		// have encountered in BFS traversal. next will denote the (partially)
+		// formed next layer.
+
+		for len(current) > 0 {
+			// Each layer adds 1 to the farthest node distance from SourceNode
+			maxdist++
+			next := []*channeldb.LightningNode{}
+
+			for _, node := range current {
+
+				// Run through all channels to find neighbours
+				if err := node.ForEachChannel(tx, func(tx *bolt.Tx,
+					edge *channeldb.ChannelEdgeInfo, outEdge, inEdge *channeldb.ChannelEdgePolicy) error {
+
+					// Get a neighbour and it's key in string format
+					neighbour := outEdge.Node
+					neighbourKey := hex.EncodeToString(neighbour.PubKey.SerializeCompressed())
+
+					if _, ok := visited[neighbourKey]; !ok {
+						// Mark this node as visited(so we don't process it again)
```

Comment:
Nit: missing a space before the paren, and a period at the end of the sentence (several other comments in the PR are missing this).